sprite.go

```diff
@@ -27,18 +27,45 @@ func (s *SpriteSheet) Release() {
 }
 
 // DrawSprite draws a sprite from the sprite sheet at the given position.
-func (s *SpriteSheet) DrawSprite(col, row uint, pos Position, flip bool) {
+func (s *SpriteSheet) DrawSprite(col, row uint, pos Position, scale float32, flip bool) {
 	src := Rectangle{
 		Pos: Position{
 			int(s.frameSize.W) * int(col),
 			int(s.frameSize.H) * int(row),
 		},
 		Size: s.frameSize,
 	}
+	scaledWidth := int(float32(s.frameSize.W) * scale)
+	scaledHeight := int(float32(s.frameSize.H) * scale)
+
+	dest := Rectangle{
+		Pos: Position{
+			X: pos.X,
+			Y: pos.Y - (scaledHeight - s.frameSize.H),
```

Comment:
This is a undocumented hack. As you are assuming the caller gives the position of top-left corner of the sprite. And you are changing the vertical position to reduce it by keeping the lower boundary intact. 

It is preferable to change the sprite API to consider that the sprite position is the bottom-center pixel of the sprite. So what the `Actor.costumePosition` function does will be done here.